server/src/test/java/org/opensearch/cluster/node/DiscoveryNodeRoleGenerator.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.node;
+
+public class DiscoveryNodeRoleGenerator {
+
+    public static DiscoveryNodeRole createUnknownRole(String roleName) {
+        return new DiscoveryNodeRole.UnknownRole(roleName, roleName, false);
```

Comment:
I think `UnknownRole`, aka `NamedRole` should have a constructor that takes the `roleName`, and we don't need a generator.

Reply:
sure, will add a public constructor.

Reply:
If we remove this generator , we need to make the `DynamicRole` class public and expose the constructor as public. That seems open some door to hack node roles. For example, some plugin can create new `DynamicRole` in code directly rather than reply on the `opensearch.yml`, that breaks the design.

This generator is only for testing. How about we keep this?